python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -0,0 +1,250 @@
+from dagster._core.execution.context.compute import OpExecutionContext
+from dagster._core.pipes.client import (
+    PipesClientCompletedInvocation,
+    PipesContextInjector,
+    PipesMessageReader,
+)
+from dagster._core.pipes.utils import PipesEnvContextInjector, open_pipes_session
+
+from dagster_aws.pipes.message_readers import PipesCloudWatchMessageReader
+
+
+@experimental
+class PipesECSClient(PipesClient, TreatAsResourceParam):
+    """A pipes client for running AWS ECS tasks.
+
+    Args:
+        client (Optional[boto3.client]): The boto ECS client used to launch the ECS task
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into the ECS task. Defaults to :py:class:`PipesEnvContextInjector`.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the ECS task. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        forward_termination (bool): Whether to cancel the ECS task when the Dagster process receives a termination signal.
+    """
+
+    def __init__(
+        self,
+        client: Optional[boto3.client] = None,
+        context_injector: Optional[PipesContextInjector] = None,
+        message_reader: Optional[PipesMessageReader] = None,
+        forward_termination: bool = True,
+    ):
+        self._client = client or boto3.client("ecs")
+        self._context_injector = context_injector or PipesEnvContextInjector()
+        self._message_reader = message_reader or PipesCloudWatchMessageReader()
+        self.forward_termination = check.bool_param(forward_termination, "forward_termination")
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    def run(
+        self,
+        *,
+        context: OpExecutionContext,
+        task_definition: str,
+        extras: Optional[Dict[str, Any]] = None,
+        count: int = 1,
+        launch_type: Optional[Literal["EC2", "FARGATE", "EXTERNAL"]] = None,
+        cluster: Optional[str] = None,
+        group: Optional[str] = None,
+        network_configuration: Optional[Dict[str, Any]] = None,
+        overrides: Optional[Dict[str, Any]] = None,
+        placement_constraints: Optional[List[Dict[str, Any]]] = None,
+        placement_strategy: Optional[List[Dict[str, Any]]] = None,
+        platform_version: Optional[str] = None,
+        reference_id: Optional[str] = None,
+        started_by: Optional[str] = None,
+        tags: Optional[List[Dict[str, str]]] = None,
+        client_token: Optional[str] = None,
+        volume_configurations: Optional[List[Dict[str, Any]]] = None,
+        capacity_provider_strategy: Optional[List[Dict[str, Any]]] = None,
+        enable_ecs_managed_tags: Optional[bool] = None,
+        enable_execute_command: Optional[bool] = None,
+        propagate_tags: Optional[Literal["TASK_DEFINITION", "SERVICE", "NONE"]] = None,
```

Comment:
I'm concerned here about keeping the API here in sync with all the ecs options and versions. Most of these just get packaged in a params dictionary. How about we just pass through params dictionary direclty?

Reply:
We can do that

Reply:
I pushed a commit with my proposal. 


```python

class RunECSTaskParams(TypedDict):
    taskDefinition: str
    count: int
    launchType: Optional[Literal["EC2", "FARGATE", "EXTERNAL"]]
    cluster: Optional[str]
    group: Optional[str]
    networkConfiguration: Optional[Dict[str, Any]]
    overrides: Optional[Dict[str, Any]]
    placementConstraints: Optional[List[Dict[str, Any]]]
    placementStrategy: Optional[List[Dict[str, Any]]]
    platformVersion: Optional[str]
    referenceId: Optional[str]
    startedBy: Optional[str]
    tags: Optional[List[Dict[str, str]]]
    clientToken: Optional[str]
    volumeConfigurations: Optional[List[Dict[str, Any]]]
    capacityProviderStrategy: Optional[List[Dict[str, Any]]]
    enableEcsManagedTags: Optional[bool]
    enableExecuteCommand: Optional[bool]
    propagateTags: Optional[Literal["TASK_DEFINITION", "SERVICE", "NONE"]]

    def run(
        self,
        *,
        context: OpExecutionContext,
        extras: Optional[Dict[str, Any]] = None,
        **params: Unpack[RunECSTaskParams],
    ) -> PipesClientCompletedInvocation:

```